tests/Integration/Behaviour/Features/Scenario/Order/order_currency.feature

```diff
@@ -0,0 +1,235 @@
+      | type      | percent               |
+      | value     | 10                    |
+    Then order "bo_order1" should have 1 cart rule
+    Then order "bo_order1" should have cart rule "discount ten-percents" with amount "€23.80"
+    Then order "bo_order1" should have following details:
+      | total_products           | 238.00 |
+      | total_products_wt        | 252.28 |
+      | total_discounts_tax_excl | 23.80  |
+      | total_discounts_tax_incl | 25.23  |
+      | total_paid_tax_excl      | 284.20 |
+      | total_paid_tax_incl      | 301.25 |
+      | total_paid               | 301.25 |
+      | total_paid_real          | 0.0    |
+      | total_shipping_tax_excl  | 70.00  |
+      | total_shipping_tax_incl  | 74.20  |
+
+  Scenario: Add cart rule of type 'free-shipping' to an order with secondary currency
+    Given I add discount to order "bo_order1" with following details:
+      | name      | discount free-shipping |
+      | type      | free_shipping          |
+    Then order "bo_order1" should have 1 cart rule
+    Then order "bo_order1" should have cart rule "discount free-shipping" with amount "€70.00"
+    Then order "bo_order1" should have following details:
+      | total_products           | 238.00 |
+      | total_products_wt        | 252.28 |
+      | total_discounts_tax_excl | 70.00  |
+      | total_discounts_tax_incl | 74.20  |
+      | total_paid_tax_excl      | 238.00 |
+      | total_paid_tax_incl      | 252.28 |
+      | total_paid               | 252.28 |
+      | total_paid_real          | 0.00   |
+      | total_shipping_tax_excl  | 70.00   |
+      | total_shipping_tax_incl  | 74.20   |
+
+  Scenario: Add cart rule of type 'amount' to an order with secondary currency and a product with specific price
+    Given there is a product in the catalog named "Test Product With Discount and SpecificPrice" with a price of 16.0 and 100 items in stock
+    And product "Test Product With Discount and SpecificPrice" has a specific price named "discount25" with a discount of 25.0 percent
+    And product "Test Product With Discount and SpecificPrice" should have specific price "discount25" with following settings:
+      | price          | -1         |
+      | from_quantity  | 1          |
+      | reduction      | 0.25       |
+      | reduction_type | percentage |
+      | reduction_tax  | 1          |
+    And there is a cart rule named "CartRuleAmountOnSelectedProduct" that applies an amount discount of 1.0 with priority 1, quantity of 100 and quantity per user 100
+    And cart rule "CartRuleAmountOnSelectedProduct" has no discount code
+    And cart rule "CartRuleAmountOnSelectedProduct" is restricted to product "Test Product With Discount and SpecificPrice"
+    When I add products to order "bo_order1" with new invoice and the following products details:
+      | name          | Test Product With Discount and SpecificPrice |
+      | amount        | 1                                     |
```

Comment:
Can you please add 2 products instead? Just to check that the total amounts are correctly computed in OrderDetail